### ISSUE TITLE
Change GeoJSON style_callback post-instantiation?

### ISSUE THREAD

**Issue by deeplook** [jupyter-widgets/ipyleaflet#675]
When I add a GeoJSON layer like this:

```python
m = Map(center=[38.8976, -77.0365], zoom=11)
gj = GeoJSON(data=open("my.geojson").read())
m.add_layer(gj)
```

I can change its style attribute later and the browser will apply the change:
 
```python
gj.style = {"color": "red"}  # changes color from default to red
```

It would be very useful if the same would happen when setting the `style_callback` attribute dynamically at any time later (and not only when instantiating the object) like this (setting it to a constant value only for illustration):

```python
gj.style_callback = lambda feature: {"color": "green"}
```

But this has no effect now. I see that `style_callback` is not listed in `gj.keys`which is likely one part of the reason... I'm trying to understand if this would be possible or not.

**Comment by sackh** [comment]
@martinRenou - I would like to work on this for JupyterCon sprint. Please assign this to me.

**Comment by martinRenou** [comment]
Thank you!

**Comment by sackh** [comment]
Below is my analysis of this issue:

1. If we first set `style_callback` then it works as expected:

![style_callback](https://user-images.githubusercontent.com/8600927/96338069-3767d080-10a9-11eb-8222-738426c2c9fc.gif)

2. If we first set `style` trait and then `style_callback` then style will not change:

![style_callback2](https://user-images.githubusercontent.com/8600927/96338091-59f9e980-10a9-11eb-84c9-e30108b6b43f.gif)

3. If we first set `style` trait and then `data` trait style will not change:

![style_callback3](https://user-images.githubusercontent.com/8600927/96338312-b3aee380-10aa-11eb-901c-fe8197538042.gif)

so when we set `style`  trait it sets style in the javascript side. but when we set `style_callback` trait it updates data.
so once style trait is updated even if you update the style in data's properties then style of geojson does not change.
It looks like something from leaflet.js.after setting style trait and then changing data does not update style of the layer.

@martinRenou  - can you help ?

**Comment by martinRenou** [comment]
I start to think this is not an issue.

The `style` property has precedence over the `style_callback` property. So if the same attribute (e.g. `color`) appears in both `style` and `style_callback`, then it's the one from `style` that is taken into account.

We should maybe just properly document this.

**Comment by sackh** [comment]
one more thing to note is that when we change the data trait of the GeoJSON we keep the same object and just clear the layer and add new data in the javascript side, as the object is the same it's style is not changed even though we update data?

**Comment by sackh** [comment]
There is a similar issue https://github.com/jupyter-widgets/ipyleaflet/issues/607, where style attributes are different still style, does not change.

**Comment by martinRenou** [comment]
So to be clear:
- `style_callback` actually changes the data
- `style` **does not** change the data, it overrides any style that was set on the data during the display

**Comment by sackh** [comment]
Thanks,  @martinRenou understood. I think we can close this one as well :)

**Comment by martinRenou** [comment]
Do you think this behavior makes sense? Should we rename those properties to make it clearer what they do?

Maybe it only needs extra documentation.

**Comment by sackh** [comment]
Yes will need to document these somewhere and yes definitely need to rename `style_callback` as it's not related to style but the data of the GeoJSON layer.

**Comment by martinRenou** [comment]
> yes definitely need to rename style_callback as it's not related to style but the data of the GeoJSON layer

Makes sense! Can you think of a better name?

**Comment by deeplook** [comment]
> Makes sense! Can you think of a better name?

Something like `load_data`?

**Comment by martinRenou** [comment]
:S This does not really convey this idea that it changes the features style in the data.

**Comment by sackh** [comment]
naming this one is a bit tricky as it updates style dict but inside GeoJSON data.

**Comment by deeplook** [comment]
`load_data_style`?
